assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,593 @@
+mk
+#### Ex. 17: Using the `{,}` construct create five directories named 
+####         `dir_A`, `dir_C`, `dir_E`, `dir_G`, and `dir_I`, and list them
+####         (but not their contents) when you are done. (Think `ls -d`).
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ mkdir {dir_A,dir_C,dir_E,dir_G,dir_I}
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls
+bam             dir_A  dir_E  dir_I  metadata  silly
+count_files.sh  dir_C  dir_G  fastq  sam       unix-intro-TEMPLATE.sh
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$
+
+#### Ex. 18: Using globbing, remove `dir_A`, `dir_C`, `dir_E`, `dir_G`, and `dir_I`,
+####         using a command with 8 characters (including spaces)
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ rmdir {dir_A,dir_C,dir_E,dir_G,dir_I}
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls
+bam  count_files.sh  fastq  metadata  sam  silly  unix-intro-TEMPLATE.sh
+
+#### Ex. 19: Create a directory called `nice_dir_name` and then use `echo`,
+####         redirecting its output to create a file called `nice_file_name`
+####         inside of `nice_dir_name` whose contents is the string
+####         `A string in a file!`
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ mkdir nice_dir_name
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls
+bam             fastq     nice_dir_name  silly
+count_files.sh  metadata  sam            unix-intro-TEMPLATE.sh
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ cd nice_dir_name
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ touch nice_file_name
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ ls
+nice_file_name
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ echo "A string in a fil
+e" > nice_file_name
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ cat nice_file_name
+A string in a file
+
+#### Ex. 20: Catenate the contents of `nice_dir_name/nice_file_name` to _stdout_
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ cat nice_dir_name
+cat: nice_dir_name: Is a directory
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ cat nice_file_name
+A string in a file
+
+#### Ex. 21: Create a directory called `bad directory name with spaces`, and then,
+####         with redirection, create a file inside it, named 
+####         `bad file name with spaces` whose contents are the string
+####         `Whoa! No spaces, please!`. Use backslash escaping instead of quoting
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ mkdir "bad directory name with spaces"
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls
+bad directory name with spaces  fastq          sam
+bam                             metadata       silly
+count_files.sh                  nice_dir_name  unix-intro-TEMPLATE.sh
+
+(base) [zhimei@colostate.edu@login11 bad] touch "bad file name with spaces"
+(base) [zhimei@colostate.edu@login11 bad] cat  "Whoa! No spaces, please!" > bad file name with spaces
```

Comment:
THe spaces in `> bad file name with spaces` must have made this do funny things.